include/netlink-private/types.h

```diff
@@ -323,6 +325,31 @@ struct rtnl_nexthop
 	struct rtnl_nh_encap *	rtnh_encap;
 };
 
+struct rtnl_nexthop_group
+{
+	struct nl_list_head	nexthop_list;
+	int			ce_refcnt;
+	uint32_t id;
+	uint8_t weight;
+};
+
+struct rtnl_nh
+{
+	NLHDR_COMMON
+
+	uint8_t nh_family;
+	uint8_t nh_scope;
+	uint8_t nh_protocol;
+	uint32_t nh_flags;
+
+	uint32_t nh_id;
+	uint32_t nh_group_type;
+	nl_nh_group_t * nh_group;
+	uint32_t nh_oif;
+	struct nl_addr * nh_gateway;
+	uint32_t nh_master;
```

Comment:
some of those fields are unused (not yet implemented?). For example, `nh_protocol`.

It would seem clearer to just drop those. Otherwise, it prompts questions like whether `nh_clone()` needs to take care of those fields.

Reply:
done